react/src/pages/VariantQueryPage.tsx

```diff
@@ -89,8 +91,13 @@ const ErrorText: React.FC<{ error?: string }> = ({ error }) => (
         </Typography>
     </ErrorWrapper>
 );
-
 const VariantQueryPage: React.FC<{}> = () => {
+    const [variants, setVariants] = useState<Array<VariantQueryDataResult> | undefined>(undefined);
+    const [searchLoading, setSearchLoading] = useState<boolean>(false);
+    const [queryType, setQueryType] = useState<'slurm' | 'osmp'>('osmp');
+
+    const [id, setId] = useState<string>('');
```

Comment:
I still need a way to somehow pass the `jobId` from Slurm to the front-end. Any ideas?

Reply:
Yeah, so the way this works is a little bit different. 

When the page loads for the first time, `useFetchVariantsSubscription` will execute a subscription and starts listening for data from the server. If the server publishes data for the first time, `useFetchVariantsSubscription` will receive it. However, if the server publishes data for the second time, `useFetchVariantsSubscription` will actually still use the old data. The way Apollo subscription hook will "resubscribe" is when we pass in a different value in `variables`. That's why I'm passing a random number to the variables to trigger re-subscription. But ideally, we'd want to know the `jobId` in advance, which I haven't figured out how to do yet.